### PR TITLE
[SCB-1450] Microservice instance may get stuck when watch connection to sc is recovered and pulling instance action is executed in eventloop thread

### PR DESCRIPTION
**Pull request by HuaiJoe**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java [hunk 1]
```diff
@@ -109,22 +109,23 @@ public HttpVersion getHttpVersion() {
     return HttpVersion.valueOf(property.get());
   }
 
-  public int getWorkerPoolSize() {
-    String workerPoolSizeKey = "servicecomb.service.registry.client.workerPoolSize";
+  public int getInstances() {
+    String instances = "servicecomb.service.registry.client.instances ";
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java]: Define a constant is better

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/AbstractClientPool.java [hunk 2]
```diff
@@ -42,27 +45,41 @@
 
   public static final String PROXY_KEY = "sc.consumer";
 
+  private static final String EVENT_LOOP_POOL_SIZE = "servicecomb.service.registry.client.eventLoopPoolSize";
+
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/AbstractClientPool.java]: move all configurations to ServiceRegistryConfig is better

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/AbstractClientPool.java [hunk 3]
```diff
@@ -42,27 +45,41 @@
...
+  private static final String WORKER_POOL_SIZE = "servicecomb.service.registry.client.workerPoolSize";
```
Comment (yhs0092) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/AbstractClientPool.java]: `servicecomb.service.registry.client.workerPoolSize` is used as the key of another configurations. This is a change not compatible. Could you update the corresponding docs?

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java (hunk 1 above)
Comment (yhs0092) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java]: There is a redundant space char at the tail of this string

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java [hunk 4]
```diff
@@ -109,22 +109,23 @@ public HttpVersion getHttpVersion() {
...
+            DynamicPropertyFactory.getInstance()
+                    .getIntProperty(instances, 1);
+    int deployInstances = property.get();
+    if (deployInstances <= 0) {
       int nAvailableProcessors = Runtime.getRuntime().availableProcessors();
       LOGGER.warn("The property `{}` must be positive integer, fallback to use number of available processors: {}",
-          workerPoolSizeKey,
-          nAvailableProcessors);
+              instances,
+              nAvailableProcessors);
```
Comment (yhs0092) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/config/ServiceRegistryConfig.java]: The indent of some files seems not consistent with the other code, like here and `AbstractClientPool`. Could you reformat your pull request? You can find the code template [here](https://github.com/apache/servicecomb-java-chassis/tree/master/etc).

**Comment by liubao68** [comment]
Can you please check if config-cc has the same problem?

**Comment by HuaiJoe** [comment]
> Can you please check if config-cc has the same problem?

Config Center client may come across the same problem as Service Center client.I will create a issue and fix it later.

**Pull request merged**